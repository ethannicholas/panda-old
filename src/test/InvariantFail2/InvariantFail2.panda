class Test {
    var i:Int := 40
    @invariant(i >= 10)

    method set(i:Int) {
        self.i := i
    }
}

class Test2 : Test {
    var j:Int := 60
    @invariant(j <= 100)
    
    @override
    method set(i:Int) {
        super.set(i)
        self.j := i
    }
}

var t := Test2()
Console.writeLine("\{t.i} - \{t.j}")
t.set(30)
Console.writeLine("\{t.i} - \{t.j}")
t.set(20)
Console.writeLine("\{t.i} - \{t.j}")
t.set(10)
Console.writeLine("\{t.i} - \{t.j}")
try {
    t.set(0)
}
catch e:AssertionError {
    Console.writeLine("assertion error")
}