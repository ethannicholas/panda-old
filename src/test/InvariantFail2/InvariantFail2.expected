40 - 60
30 - 30
20 - 20
10 - 10
assertion error
